src/raftstore/store/snap.rs

```diff
@@ -618,6 +620,1006 @@ mod v1 {
+                let env_opt = EnvOptions::new();
+                let io_options = Options::new();
+                let mut writer = SstFileWriter::new(&env_opt, &io_options);
+                if let Err(e) = writer.open(&cf_file.tmp_path) {
+                    return Err(io::Error::new(ErrorKind::Other, e));
+                }
+                cf_file.sst_writer = Some(writer);
+            }
+            let f = try!(OpenOptions::new()
+                .write(true)
+                .create_new(true)
+                .open(&self.meta_file.tmp_path));
+            self.meta_file.file = Some(f);
+            Ok(())
+        }
+
+        fn init_for_sending(&mut self) -> RaftStoreResult<()> {
+            if !self.exists() {
+                return Err(box_err!("snapshot file {} not exist", self.path()));
+            }
+            for cf_file in &mut self.cf_files {
+                // initialize cf file size and reader
+                let size = try!(get_file_size(&cf_file.path));
+                let file = try!(File::open(&cf_file.path));
+                cf_file.size = size;
+                cf_file.file = Some(file);
+            }
+            self.load_snapshot_metadata()
+        }
+
+        fn init_for_receiving(&mut self) -> io::Result<()> {
+            if self.exists() || self.meta_file.file.is_some() {
+                return Ok(());
+            }
+
+            for cf_file in &mut self.cf_files {
+                let f =
+                    try!(OpenOptions::new().write(true).create_new(true).open(&cf_file.tmp_path));
+                cf_file.file = Some(f);
+            }
+            let f = try!(OpenOptions::new()
+                .write(true)
+                .create_new(true)
+                .open(&self.meta_file.tmp_path));
+            self.meta_file.file = Some(f);
+            Ok(())
+        }
+
+        fn init_for_applying(&mut self) -> RaftStoreResult<()> {
+            self.load_snapshot_metadata()
+        }
+
+        fn load_snapshot_metadata(&mut self) -> RaftStoreResult<()> {
+            let size = try!(get_file_size(&self.meta_file.path));
+            let mut file = try!(File::open(&self.meta_file.path));
+            let mut buf = Vec::with_capacity(size as usize);
+            try!(file.read_to_end(&mut buf));
+            let mut snapshot_data = RaftSnapshotData::new();
+            try!(snapshot_data.merge_from_bytes(&buf));
+            self.meta_file.data = snapshot_data;
+            Ok(())
+        }
+
+        fn get_display_path(dir_path: &PathBuf, prefix: &str) -> String {
+            let mut cf_names = String::from("");
+            for (i, cf) in get_snapshot_cfs().iter().enumerate() {
+                if i == 0 {
+                    cf_names += "(";
+                    cf_names += cf;
+                } else {
+                    cf_names += "|";
+                    cf_names += cf;
+                }
+            }
+            cf_names += ")";
+            format!("{}/{}_{}{}",
```

Comment:
I suggest using a unique directory to save SST files of the snapshot. So all may like:

```
region_index_term_recv_sst
region_index_term_recv_sst/lock.sst
region_index_term_recv_sst/write.sst
region_index_term_recv_sst/default.sst
region_index_term_send_sst
......
```

So you can use the directory name for the snapshot path name.